error/s2n_errno.h

```diff
@@ -156,6 +156,15 @@ typedef enum {
     S2N_ERR_CANCELLED,
     S2N_ERR_INVALID_SERIALIZED_SESSION_STATE,
     S2N_ERR_SERIALIZED_SESSION_STATE_TOO_LONG,
+    S2N_ERR_CLIENT_AUTH_NOT_SUPPORTED_IN_SESSION_TICKET_MODE,
```

Comment:
This is valid in both cases of session resumption, should we re-name this to S2N_ERR_CLIENT_AUTH_NOT_SUPPORTED_IN_SESSION_RESUMPTION_MODE?

Reply:
Okay. Will update it.